src/Channels/Memory.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+namespace Channels
+{
+    public struct Memory<T>
```

Comment:
I will try to change Bytes to Bytes<T>. Separately we should talk about the best names for these types. I am not sure Memory<T> (or Bytes<T>) is the best name. It's a very generic name for a type that will probably be used in specific and advanced scenarios.